### PR TITLE
Refactor api/users_controller specs

### PR DESCRIPTION
**Pull request by athal7**

### CODE REVIEW

**Comment by jhawthorn** [comment]
:+1:

File: api/spec/controllers/spree/api/users_controller_spec.rb [hunk 1]
```diff
@@ -9,145 +9,164 @@ module Spree
...
+
+          expect(Spree.user_class.count).to eq 3
+          expect(json_response['count']).to eq 1
+          expect(json_response['users'].size).to eq 1
+        end
+
+        context "with the ability to display users" do
+          before do
+            user.spree_roles.create!(name: "user_display")
+            Spree::RoleConfiguration.configure do |config|
```
Comment (Senjai) [inline api/spec/controllers/spree/api/users_controller_spec.rb]: Is there a chance of this leaking? This appears to me to be global configuration, we'd have to reset it after the test.

Comment (athal7) [inline api/spec/controllers/spree/api/users_controller_spec.rb]: hm yea. you have more knowledge here. would it be beneficial to clear these after every test run like we do for preferences? or is state required for some specs.

Comment (Senjai) [inline api/spec/controllers/spree/api/users_controller_spec.rb]: You can either stub the current ability, with your own ability that you can manually `activate!` a permission set directly on. Or you can reset `Spree::RoleConfiguration.instance.roles` to the empty hash it was. Though it was a 'custom'.

I generally prefer:

```ruby
my_special_ability = Spree::Ability.new(user)
Spree::PermissionSets::UserDisplay.new(my_special_ability).activate!
allow(controller).to receive(:current_ability).and_return(my_special_ability)
```

Comment (athal7) [inline api/spec/controllers/spree/api/users_controller_spec.rb]: :+1:  thanks for the suggestions. will update

File: api/spec/controllers/spree/api/users_controller_spec.rb [hunk 2]
```diff
@@ -9,145 +9,164 @@ module Spree
...
+          api_get :new, token: user.spree_api_key
+          expect(json_response["attributes"]).to eq(attributes.map(&:to_s))
+        end
+      end
+
+      describe "#create" do
+        it "can create a new user" do
+          user_params = { email: 'new@example.com', password: 'spree123', password_confirmation: 'spree123' }
+
+          api_post :create, :user => user_params, token: user.spree_api_key
```
Comment (athal7) [inline api/spec/controllers/spree/api/users_controller_spec.rb]: note to self - this hash syntax mixing is sloppy

Comment (BenMorganIO) [inline api/spec/controllers/spree/api/users_controller_spec.rb]: I'm :+1: for 1.9 syntax

**Comment by athal7** [comment]
closing, going to open as part of different PR

**Pull request closed without merging**